CMakeLists.txt.ros2

```diff
@@ -0,0 +1,263 @@
+find_package(sensor_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(visualization_msgs REQUIRED)
+
+find_package(common_robotics_utilities REQUIRED)
+
+find_package(Eigen3 REQUIRED)
+set(Eigen3_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
+find_package(OpenMP)
+find_package(CUDA)
+find_package(OpenCL)
+
+#######################################
+## Declare ROS messages and services ##
+#######################################
+
+## Generate messages in the 'msg' folder
+
+rosidl_generate_interfaces(
+  ${PROJECT_NAME}_interfaces
+  msg/CollisionMapMessage.msg
+  msg/DynamicSpatialHashedCollisionMapMessage.msg
+  msg/SignedDistanceFieldMessage.msg
+  msg/TaggedObjectCollisionMapMessage.msg
+  DEPENDENCIES std_msgs
+)
+
+###########
+## Build ##
+###########
+
+## Specify additional locations of header files
+## Your package locations should be listed before other locations
+include_directories(include SYSTEM ${Eigen3_INCLUDE_DIRS})
+
+## Build options
+add_compile_options(-std=c++14)
+add_compile_options(-Wall)
+add_compile_options(-Wextra)
+add_compile_options(-Werror)
+add_compile_options(-Wconversion)
+add_compile_options(-Wshadow)
+add_compile_options(-O3)
+add_compile_options(-g)
+add_compile_options(-flto)
+add_compile_options(-Werror=non-virtual-dtor)
+add_compile_options(-Wold-style-cast)
+add_compile_options(-Wmaybe-uninitialized)
+add_compile_options(-march=native)
+
+add_definitions(-DVOXELIZED_GEOMETRY_TOOLS__SUPPORTED_ROS_VERSION=2)
+
+## It's not clear if add_compile_options does the right things for flags that
+## may differ between languages and target type.
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+set(CMAKE_EXE_LINKER_FLAGS
+    "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
+set(CMAKE_SHARED_LINKER_FLAGS
+    "${CMAKE_SHARED_LINKER_FLAGS} ${OpenMP_SHARED_LINKER_FLAGS}")
+
+# Voxelized geometry tools library
+add_library(${PROJECT_NAME}
+            include/${PROJECT_NAME}/collision_map.hpp
+            include/${PROJECT_NAME}/dynamic_spatial_hashed_collision_map.hpp
+            include/${PROJECT_NAME}/signed_distance_field.hpp
+            include/${PROJECT_NAME}/signed_distance_field_generation.hpp
+            include/${PROJECT_NAME}/tagged_object_collision_map.hpp
+            include/${PROJECT_NAME}/topology_computation.hpp
+            src/${PROJECT_NAME}/collision_map.cpp
+            src/${PROJECT_NAME}/dynamic_spatial_hashed_collision_map.cpp
+            src/${PROJECT_NAME}/tagged_object_collision_map.cpp)
+ament_target_dependencies(${PROJECT_NAME}
+                          common_robotics_utilities)
+ament_target_dependencies(${PROJECT_NAME} SYSTEM
+                          geometry_msgs
+                          rclcpp
+                          std_msgs
+                          visualization_msgs)
+add_dependencies(${PROJECT_NAME} ${PROJECT_NAME}_interfaces)
+target_include_directories(${PROJECT_NAME}
```

Comment:
This library shouldn't need any ROS dependencies beyond `common_robotics_utilities`

Reply:
Indeed. I blindly replaced `${catkin_LIBRARIES}` in the ROS 1 `CMakeLists.txt`. See https://github.com/calderpg/voxelized_geometry_tools/pull/6/commits/d0ab09ff01db86b614970dfa67a127a521e90fae.

Reply:
I believe it shouldn't need to include the generated message headers either?

Reply:
That is also true, and it applies to the ROS 1 `CMakeLists.txt` as well.